HarClient.Tests/Validation/HarAssert.cs

```diff
@@ -0,0 +1,251 @@
+                AssertIsValidPostData(o);
+            AssertIsNumberOrMinusOne(request, "headersSize");
+            AssertIsNumberOrMinusOne(request, "bodySize");
+            _ = HasOptionalComment(request);
+
+            static void AssertIsValidPostData(JObject postData)
+            {
+                _ = JAssert.HasStringProperty(postData, "mimeType");
+                foreach (var item in JAssert.HasArrayProperty(postData, "params"))
+                {
+                    var p = JAssert.IsObject(item);
+                    _ = JAssert.HasStringProperty(p, "name");
+                    _ = JAssert.HasOptionalStringProperty(p, "value");
+                    _ = JAssert.HasOptionalStringProperty(p, "fileName");
+                    _ = JAssert.HasOptionalStringProperty(p, "contentType");
+                    _ = HasOptionalComment(p);
+                }
+                _ = JAssert.HasStringProperty(postData, "text");
+                _ = HasOptionalComment(postData);
+            }
+        }
+        public static void IsValidResponse(Response response) => AssertIsValidResponse(response.ToJObject());
+        static void AssertIsValidResponse(JObject response)
+        {
+            _ = JAssert.HasProperty(response, "status", JTokenType.Integer);
+            _ = JAssert.HasStringProperty(response, "statusText");
+            _ = JAssert.HasStringProperty(response, "httpVersion");
+            AssertIsValidCookies(JAssert.HasArrayProperty(response, "cookies"));
+            AssertIsValidRecords(JAssert.HasArrayProperty(response, "headers"));
+            AssertIsValidContent(JAssert.HasObjectProperty(response, "content"));
+            _ = JAssert.HasStringProperty(response, "redirectURL");
+            AssertIsNumberOrMinusOne(response, "headersSize");
+            AssertIsNumberOrMinusOne(response, "bodySize");
+            _ = HasOptionalComment(response);
+
+            static void AssertIsValidContent(JObject content)
+            {
+                _ = JAssert.HasProperty(content, "size", JTokenType.Integer);
+                _ = JAssert.HasOptionalProperty(content, "compression", JTokenType.Integer);
+                _ = JAssert.HasStringProperty(content, "mimeType");
+                _ = JAssert.HasOptionalStringProperty(content, "text");
+                _ = HasOptionalComment(content);
+            }
+        }
+        static void AssertIsValidCookies(JArray cookies)
+        {
+            foreach (var item in cookies)
+            {
+                var cookie = JAssert.IsObject(item);
+                _ = JAssert.HasStringProperty(cookie, "name");
+                _ = JAssert.HasStringProperty(cookie, "value");
+                _ = JAssert.HasOptionalStringProperty(cookie, "path");
+                _ = JAssert.HasOptionalStringProperty(cookie, "domain");
+                if (JAssert.HasOptionalStringProperty(cookie, "expires") is string s)
+                    StringAssert.Matches(s, isoDatetime);
```

Comment:
Match with datetime is used in a few spots. Consider refactoring into a convenience method.

Reply:
YAGNI